### PR TITLE
LibJS: Temporal fixes

### PR DESCRIPTION
**Pull request by linusg**
```
Diff Tests:
    test/built-ins/Temporal/Instant/compare/instant-string-multiple-offsets.js                                                                                              ❌ -> ✅
    test/built-ins/Temporal/Instant/compare/instant-string.js                                                                                                               ❌ -> ✅
    test/built-ins/Temporal/Instant/from/instant-string-multiple-offsets.js                                                                                                 ❌ -> ✅
    test/built-ins/Temporal/Instant/from/instant-string.js                                                                                                                  ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/equals/instant-string-multiple-offsets.js                                                                                     ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/since/instant-string-multiple-offsets.js                                                                                      ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/since/instant-string.js                                                                                                       ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/toJSON/basic.js                                                                                                               ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/toString/timezone-string-datetime.js                                                                                          ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/toString/timezone-string-multiple-offsets.js                                                                                  ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/toZonedDateTime/timezone-string-datetime.js                                                                                   ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/toZonedDateTimeISO/timezone-string-datetime.js                                                                                ❌ -> ✅
    test/built-ins/Temporal/Instant/prototype/until/instant-string-multiple-offsets.js                                                                                      ❌ -> ✅
    test/built-ins/Temporal/Now/zonedDateTime/timezone-string-datetime.js                                                                                                   ❌ -> ✅
    test/built-ins/Temporal/Now/zonedDateTimeISO/timezone-string-datetime.js                                                                                                ❌ -> ✅
    test/built-ins/Temporal/PlainDate/prototype/toZonedDateTime/timezone-string-datetime.js                                                                                 ❌ -> ✅
    test/built-ins/Temporal/PlainDateTime/prototype/toZonedDateTime/timezone-string-datetime.js                                                                             ❌ -> ✅
    test/built-ins/Temporal/PlainTime/prototype/toZonedDateTime/timezone-string-datetime.js                                                                                 ❌ -> ✅
    test/built-ins/Temporal/TimeZone/from/argument-valid.js                                                                                                                 ❌ -> ✅
    test/built-ins/Temporal/TimeZone/from/timezone-string-datetime.js                                                                                                       ❌ -> ✅
    test/built-ins/Temporal/TimeZone/prototype/getPlainDateTimeFor/instant-string-multiple-offsets.js                                                                       ❌ -> ✅
    test/built-ins/Temporal/TimeZone/prototype/getPlainDateTimeFor/pre-epoch.js                                                                                             ❌ -> ✅
    test/built-ins/Temporal/ZonedDateTime/from/timezone-string-datetime.js                                                                                                  ❌ -> ✅
    test/built-ins/Temporal/ZonedDateTime/prototype/equals/timezone-string-multiple-offsets.js                                                                              💥️ -> ❌
    test/built-ins/Temporal/ZonedDateTime/prototype/since/timezone-string-multiple-offsets.js                                                                               💥️ -> ✅
    test/built-ins/Temporal/ZonedDateTime/prototype/until/timezone-string-multiple-offsets.js                                                                               💥️ -> ✅
    test/built-ins/Temporal/ZonedDateTime/prototype/withTimeZone/subclassing-ignored.js                                                                                     ❌ -> ✅
    test/built-ins/Temporal/ZonedDateTime/prototype/withTimeZone/timezone-string-datetime.js                                                                                ❌ -> ✅
    test/built-ins/Temporal/ZonedDateTime/timezone-string-datetime.js                                                                                                       ❌ -> ✅

    test/intl402/Temporal/Now/zonedDateTime/timezone-string-datetime.js                                                                                                     ❌ -> ✅
    test/intl402/Temporal/Now/zonedDateTimeISO/timezone-string-datetime.js                                                                                                  ❌ -> ✅
```

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp [hunk 1]
```diff
@@ -297,28 +297,31 @@ ThrowCompletionOr<double> parse_time_zone_offset_string(GlobalObject& global_obj
...
     }
     // 11. Else,
     else {
         // a. Let nanoseconds be 0.
         nanoseconds = 0;
     }
     // 12. Return sign × (((hours × 60 + minutes) × 60 + seconds) × 10^9 + nanoseconds).
-    return sign * (((hours * 60 + minutes) * 60 + seconds) * 1000000000 + nanoseconds);
+    // NOTE: Decimal point in 10^9 is important, otherwise it's all integers and the result overflows!
+    return sign * (((hours * 60 + minutes) * 60 + seconds) * 1000000000.0 + nanoseconds);
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp:324]: Should this be split into a separate commit similarly to the second commit?

**Review by IdanHo — COMMENTED** [review]
Looks good, just a nitpick

File: Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp:324]: No, as it only becomes relevant when changing nanoseconds from a `double` to an `i32` in the same commit. Omitting it would break the commit, and before the commit it's not relevant :^)
